sources/Engine.ts

```diff
@@ -69,17 +70,26 @@ export class Engine {
       // Ignore errors; too bad
     }
 
-    if (typeof lastKnownGood !== `object` || lastKnownGood === null)
-      return definition.default;
+    if (typeof lastKnownGood === `object` && lastKnownGood !== null &&
+        Object.prototype.hasOwnProperty.call(lastKnownGood, packageManager)) {
+      const override = (lastKnownGood as any)[packageManager];
+      if (typeof override === `string`) {
+        return override;
+      }
+    }
 
-    if (!Object.prototype.hasOwnProperty.call(lastKnownGood, packageManager))
+    if (process.env.COREPACK_NO_LOOKUP)
       return definition.default;
 
-    const override = (lastKnownGood as any)[packageManager];
-    if (typeof override !== `string`)
-      return definition.default;
+    const {[`dist-tags`]: {latest}, versions: {[latest]: {dist: {shasum}}}} = await fetchAsJson(`https://registry.npmjs.org/${packageManager}`);
```

Comment:
We can't hardcode the npm registry here, it needs to look at what `config.json` specifies.
See the following section for an example https://github.com/nodejs/corepack/blob/31913aa3172e8c3fe65528996f9b8f32a43b52a6/sources/corepackUtils.ts#L12-L26